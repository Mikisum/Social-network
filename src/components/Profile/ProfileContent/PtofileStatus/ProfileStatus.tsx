import React, {FC, useState } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { updateUsersStatus } from '../../../../redux/profileReducer'
import { AppStateType } from '../../../../redux/redux-store'
import { Typography } from 'antd';
import { EditOutlined } from '@ant-design/icons';

const { Paragraph } = Typography;

export const ProfileStatus: FC<PropsType> = ({isOwner}) => {

  const dispatch = useDispatch()

  const stateStatus = useSelector((state: AppStateType) => state.profilePage.status)

  const [status, setStatus] = useState(stateStatus);
  const updateStatus = (editVal: string) => {
    setStatus(editVal)
    dispatch(updateUsersStatus(editVal))
  }

  return (
 
      <Paragraph
        
        editable={isOwner && {
          icon: <EditOutlined   />,
          onChange: updateStatus,
          maxLength: 50,
          autoSize: { maxRows: 5, minRows: 1 },
        }}
      >
        {status || stateStatus}
      </Paragraph>
    
  );
};

type PropsType = {isOwner: boolean}
