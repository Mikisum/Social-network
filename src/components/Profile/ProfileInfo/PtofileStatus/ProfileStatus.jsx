import React from 'react'

class ProfileStatus extends React.Component {
 
  state = {
    editMode: false,
    status: this.props.status
  }

  activateEditMode = () => {
    this.setState({
      editMode: true
    })
  }

  deactivateEditMode = () => {
    this.setState({
      editMode: false
    })
    this.props.updateUsersStatus(this.state.status)
  }

  onStatusChange = (e) => {
    this.setState({
      status: e.currentTarget.value
    })
  }

  componentDidUpdate = (prevProps, prevstate) => {
    if (prevProps.status !== this.props.status)
    this.setState({
      status: this.props.status
    })
  }

  render(){
    return (
      <div>
        {
          !this.state.editMode &&
            <div>
              <span onClick={this.activateEditMode}>{this.props.status || '---'}</span>
            </div>
        }
        {
          this.state.editMode &&
            <div>
              <input 
                value={this.state.status}
                onBlur={this.deactivateEditMode}
                onChange={this.onStatusChange}
                autoFocus
              />
            </div>
        } 
      </div>
    )
  }
}

export default ProfileStatus