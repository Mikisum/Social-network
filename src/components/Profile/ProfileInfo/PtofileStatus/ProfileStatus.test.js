import React from "react";
import ProfileStatus from "./ProfileStatus.jsx";
import renderer from 'react-test-renderer';

describe("ProfileStatus component", () => {
  it('status from props should been in the state', () => {
  const component = renderer.create(<ProfileStatus status='hello my friend'/>)
  const tree = component.toJSON()
  const instance = component.getInstance()
  expect(tree).toMatchSnapshot()
  })
})