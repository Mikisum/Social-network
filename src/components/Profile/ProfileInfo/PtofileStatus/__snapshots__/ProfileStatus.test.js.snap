// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileStatus component status from props should been in the state 1`] = `
<div>
  <div>
    <span
      onClick={[Function]}
    >
      hello my friend
    </span>
  </div>
</div>
`;
